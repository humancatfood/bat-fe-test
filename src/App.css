/* Feel free to edit or remove */

.bui-app {
    font-family: sans-serif;
    text-align: center;
    border-bottom: 2px solid #666;
}

.bui-app-logo {
    height: 40px;
}

.bui-app-header {
    background-color: #fff;
    height: auto;
    padding: 20px;
    color: #000;
}

.bui-app-intro {
    font-size: 1rem;
}