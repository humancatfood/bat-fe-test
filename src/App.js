/* Feel free to edit */
import React from 'react';
import { Provider as ReduxProvider } from 'react-redux';
import classnames from 'classnames';

import { Provider as GraphQLProvider, hooks as dataHooks } from './data';

import { useBookingSelector } from './data';

import configureStore from './data/store';

import { Router, Route, Switch } from './components/Routing';

import Overview from './views/Overview';
import FourOhFour from './views/FourOhFour';

import Header from './components/Header';
import BookingsTableView from './components/BookingsTableView';
import BookingDetailView from './components/BookingDetailView';



const GraphQLTest = () => {

  const res = dataHooks.useBookings('tomorrow');

  console.log('res:', res);

  return null;
};

const App = () => {

  const [bid] = useBookingSelector();

  return (
    <div className="layout">
      <GraphQLTest />
      <Header className="layout__header"/>
      <main className={ classnames('layout__body', {'has-selected': !!bid}) }>
        <Switch>
          <Route path="/"exact component={Overview} />
          <Route path="/:date(\d{4}-\d{2}-\d{2})" render={({ match }) => (
              <>
                <BookingsTableView date={match.params.date }/>
                {
                  bid && (
                    <BookingDetailView bid={bid} />
                  )
                }
              </>
          )}>
          </Route>
          <Route component={FourOhFour} />
        </Switch>
      </main>
    </div>
  );
};



const ProvidedApp = () => (
  <Router>
    <GraphQLProvider>
      <ReduxProvider store={ configureStore() }>
        <App />
      </ReduxProvider>
    </GraphQLProvider>
  </Router>
);

export default ProvidedApp;
