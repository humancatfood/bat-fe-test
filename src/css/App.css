/* Feel free to edit or remove */



.bui-app {
    text-align: center;
}

.bookings-table-list {
    list-style: none;
    margin: 0 auto;
    padding: 0;
}

.bookings-table {
    width: 100%;
    text-align: left;
}

.bookings-table thead {
    background-color: #ccc;
}

.bookings-table,
.bookings-table td,
.bookings-table th {
    border: 1px solid black;
    border-collapse: collapse;
    padding: 5px 10px;
}

.bookings-table .cancelled {
    text-decoration: line-through;
}

.bookings-table .selected {
    background-color: #666;
    color: white;
}

section h2 {
    text-align: left;
    padding: 10px 0;
}



