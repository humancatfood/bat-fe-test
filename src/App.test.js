import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';

import App from './App';



describe('App', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it(`renders as expected`, () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

});
