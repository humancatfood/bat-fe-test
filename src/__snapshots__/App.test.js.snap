// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders as expected 1`] = `
<div>
  <div
    class="layout"
  >
    <header
      class="layout__header"
    >
      <img
        alt="Bookatable by Michelin"
        class="header__img"
        src="bookatable-by-michelin-logo.svg"
      />
    </header>
    <main
      class="layout__body"
    >
      <h1>
        Overview
      </h1>
    </main>
  </div>
</div>
`;
